Issue10 create data model

hlillian246 opened the pull request:
Hi guys, I created the data model in models.py, added some requirements in requirements.txt and some packages in VagrantFile, created some tests in the test_models.py

Below are the nosetests results 

```
Test Cases for Product Model 
- Create a product and add it to the database
- Create a product and assert that it exists
- Delete a Pet
- Test deserialization of a Product
- Test deserialization of bad data
- Find Products by Category
- Find a Product by Name
- Find a Product by ID
- Test serialization of a Product
- Update a Product

REST API Server Tests 
- Test index call

----------------------------------------------------------------------
XML: /vagrant/unittests.xml
Name                  Stmts   Miss  Cover   Missing
---------------------------------------------------
service/__init__.py      25      4    84%   28, 37-40
service/models.py        62      5    92%   41, 57, 90, 122-123
service/service.py       12      0   100%
---------------------------------------------------
TOTAL                    99      9    91%
----------------------------------------------------------------------
Ran 11 tests in 1.508s

OK

yuhanzz commented:
Hi Lillian,

1. Some minor suggestions: 
(i) replace "<your resource name>" with "Product" in all the annotations
(ii) Maybe move logger = logging.getLogger("flask.app") into the Product class and then use logger = logging.getLogger(__name__) So that the log information will be more precise

2. I think in the scope of this story, what we need to do is only serialize, deserialize, and init_db. The tests and methods for CRUD should be done in other stories. I didn't review these extra code because I guess these methods might be overwritten later by the assignees. Maybe notify other teammates about what you have done concerning their stories. Anyway, thanks for your extra work lol.

hlillian246 commented:
I have made the following adjustments to the code after Yuhan's comment. 

- deleted the unneccessary code such as the tests and methods for CRUD

- replace the "<your resource name>" with "Product" in all the annotations

- moved the logger into the  Product class and used the  logger = logging.getLogger(_name_) 

These are the nosetest results of the newly updated code
```
Test Cases for Product Model 
- Test deserialization of a Product
- Test deserialization of bad data
- Test serialization of a Product

REST API Server Tests 
- Test index call

----------------------------------------------------------------------
XML: /vagrant/unittests.xml
Name                  Stmts   Miss  Cover   Missing
---------------------------------------------------
service/__init__.py      25      4    84%   28, 37-40
service/models.py        34      2    94%   41, 68
service/service.py       12      0   100%
---------------------------------------------------
TOTAL                    71      6    92%
----------------------------------------------------------------------
Ran 4 tests in 1.087s

OK
